src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -1901,22 +1909,34 @@ private static object FallbackDeserializer(BinaryTokenStreamReader stream)
                 timer.Start();
                 FallbackDeserializations.Increment();
             }
-
-            var n = stream.ReadInt();
-            var bytes = stream.ReadBytes(n);
-            var formatter = new BinaryFormatter();
-            object ret = null;
-            using (var memoryStream = new MemoryStream(bytes))
-            {
-                ret = formatter.Deserialize(memoryStream);
-            }
-
+            var retVal = GetFallbackSerializer(UseJsonFallbackSerializer).Deserialize(null, stream);
             if (timer != null)
             {
                 timer.Stop();
                 FallbackDeserTimeStatistic.IncrementBy(timer.ElapsedTicks);
             }
-            return ret;
+
+            return retVal;
+        }
+
+        private static ISerializer GetFallbackSerializer(bool useJsonSerializer)
+        {
+            ISerializer serializer;
+            if (useJsonSerializer)
+            {
+                serializer = new OrleansJsonSerializer();
+            }
+            else
+            {
+#if DNXCORE50
+                throw new OrleansException("Can't use binary formatter as fallback serializer while running on .Net Core");
+#else
+                serializer = new BinaryFormatterSerializer();
```

Comment:
Does not seem like we need to allocate a new `BinaryFormatterSerializer` instance every time it is called, right? It is pretty much stateless, just a call through into Ser/Deser/Copy methods. So better allocate once and return, save some GC.
Same applies to `OrleansJsonSerializer`.
Serialization code is of course the most sensitive to perf, as it is on the hottest path.